src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -66,52 +67,58 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
   /**
    * Iterate through workflows array and map each element to a WorkflowTableRow object.
    */
-  const workflowElements: PaginatedSearchTableRow[] = workflows.map((value) => {
-    const engine = value.engine;
-
-    let metrics = [];
-    if (value?.metrics) {
-      metrics = value.metrics.map((metric) => {
-        return {
-          metricId: metric.id,
-          name: metric.name,
-          value: metric.result?.content_serialized ?? '',
-          status: metric.result?.exec_state?.status ?? ExecutionStatus.Unknown,
-        };
-      });
-    }
+  const workflowElements: PaginatedSearchTableRow[] = workflows.map(
+    (workflow) => {
+      const engines = reduceEngineTypes(
+        workflow.engine,
+        workflow.operator_engines.filter((x) => !!x)
```

Comment:
Does a null/undefined operator engine here imply that the engine is Aqueduct?

Reply:
No, that means the engine is inheriting the dag's setting. But there's actually a bug here that we should actually map operator's engine to the dag's engine in the list, so that the conditioning in `reduceEngineTypes` is working properly